cmd/axelard/cmd/vald/evm/evm.go

```diff
@@ -702,14 +702,20 @@ func getLatestFinalizedBlockNumber(client rpc.Client, confHeight uint64) (*big.I
 		}
 
 		return header.Number.ToInt(), nil
-	default:
-		blockNumber, err := client.BlockNumber(context.Background())
-		if err != nil {
-			return nil, err
+	case rpc.Eth2Client:
+		// TODO: check error after the merge is settled on ethereum mainnet
+		finalizedHeader, _ := client.FinalizedHeader(context.Background())
+		if finalizedHeader != nil {
+			return finalizedHeader.Number, nil
 		}
```

Comment:
Error out on the `default` case for sanity?

Reply:
I kinda wanna keep it open for now cuz we are gonna come back and change it soon.

Reply:
Can't. It would make testing with mock not work anymore.